conbench/machine_info.py

```diff
@@ -80,10 +80,6 @@ def r_info():
     }
 
 
-class GitParseWarning(RuntimeWarning):
-    pass
```

Comment:
I just [commented](https://github.com/conbench/conbench/issues/668#issuecomment-1416367880) on the issue you created about this. I'd rather not rip this out in this PR, because it creates some noise in the pytest summary. Maybe you can put it back and we can address your concerns in a followup PR?

Reply:
Thanks! I added my thoughts to that PR. You are right, it's better to not touch the warnings themselves here in this PR. I have just added a pragmatic check for the local git dir instead. As described in #668 I think that's the responsible thing to do, to potentially save others time.